src/main/xerus/monstercat/downloader/SongView.kt

```diff
@@ -50,20 +51,31 @@ class SongView(private val sorter: ObservableValue<ReleaseSorting>):
 			
 			init {
 				itemProperty().listen { item ->
-					if((item as? Release)?.downloadable == false ||
-						(item is Track && (treeItem.parent.value as? Release)?.downloadable == false)) {
+					fun disableDownload(tooltipString: String) {
 						if(tooltip == null) {
 							listener.onChanged(null)
 							children.addListener(listener)
 							styleClass.add("not-downloadable")
-							tooltip = Tooltip("This Release is currently not downloadable")
+							tooltip = Tooltip(tooltipString)
 						}
-					} else {
-						if(tooltip != null) {
-							children.removeListener(listener)
-							children.filterIsInstance<CheckBox>().firstOrNull()?.isDisable = false
-							styleClass.remove("not-downloadable")
-							tooltip = null
+					}
+					when {
+						// Downloadable check
+						(item as? Release)?.downloadable == false ||
+							(item is Track && (treeItem.parent.value  as? Release)?.downloadable == false) ->
+							disableDownload("This Release is currently not downloadable")
+						
+						// Licensable check (if in streamer mode)
+						((item as? Track)?.licensable == false || (item as? Release)?.tracks?.none { it.licensable } == true)
+							&& Settings.SKIPUNLICENSABLE() -> disableDownload("This Release is not licensable")
```

Comment:
I think you should check the setting first to avoid the other more expensive checks if the Setting is false anyways

Reply:
@defvs please fix this, so we can finally get this thing merged :sweat_smile: